legend-depot-core-services/src/main/java/org/finos/legend/depot/services/projects/ProjectsServiceImpl.java

```diff
@@ -301,17 +303,51 @@ public ProjectDependencyReport buildReportFromGraph(ProjectDependencyGraph depen
     }
 
     @Override
-    public List<ProjectDependencyWithPlatformVersions> getDependentProjects(String groupId, String artifactId, String versionId)
+    public List<ProjectDependencyWithPlatformVersions> getDependantProjects(String groupId, String artifactId, String versionId)
     {
+        return getDependantProjects(groupId, artifactId, versionId, false);
+    }
+
+    @Override
+    public List<ProjectDependencyWithPlatformVersions> getDependantProjects(String groupId, String artifactId, String versionId, boolean latestOnly)
+    {
+        List<StoreProjectVersionData> projectVersionDatas;
+        if (latestOnly)
+        {
+            Map<String, List<StoreProjectVersionData>> groupedProjectVersionData = projectsVersions.getAll().stream().collect(Collectors.groupingBy(p -> p.getGroupId() + p.getArtifactId()));
+            projectVersionDatas = groupedProjectVersionData.entrySet().stream().map(set ->
+            {
+                Collections.sort(set.getValue(), (o1, o2) ->
+                {
+                    if (VersionValidator.isSnapshotVersion(o1.getVersionId()))
+                    {
+                        return -1;
+                    }
+                    else if (VersionValidator.isSnapshotVersion(o2.getVersionId()))
+                    {
+                        return -1;
+                    }
+                    else
+                    {
+                        return VersionId.parseVersionId(o2.getVersionId()).compareTo(VersionId.parseVersionId(o1.getVersionId()));
+                    }
+                });
+                return set.getValue().get(0);
+            }).collect(Collectors.toList());
+        }
+        else
+        {
+            projectVersionDatas = projectsVersions.getAll();
+        }
         if (versionId.equalsIgnoreCase("ALL"))
```

Comment:
why not move this portion to another function. The function will have an argument list of storeProjectVersionData and its function would be to do the actual checking.